Tasks/MavenAuthenticateV0/task.json

```diff
@@ -38,6 +45,15 @@
             "properties": {
                 "MultiSelectFlatList": "true"
             }
+        },
+        {
+            "name": "useAgentTempDir",
+            "type": "boolean",
+            "label": "Use agent temp directory for .m2 folder",
+            "defaultValue": "false",
+            "required": false,
+            "helpMarkDown": "If this is true, this task will use the AGENT_TEMPDIRECTORY location instead of USERPROFILE (Windows) or HOME (Linux) to store the .m2 folder. This will solve concurrency issues when running agents as the same user on the same server.",
```

Comment:
Need to update this to say that we publish a pipeline variable called userM2SettingsXmlPath which contains the location of the settings.xml file. Subsequent maven operations need to set the path for the settings file using this variable, ex.

mvn install --settings $(userM2SettingsXmlPath)

Reply:
We also don't need to provide internal details of the task here, just that we will provide this env variable.
We also would need to update public docs for this.